### PR TITLE
refactor(replication-job): refactored from bash to python

### LINKED ISSUE #1: [replication-job] container not crashing when osmosis fails
Because of the way the container start script is currently written, when osmosis fails for any reason, the script keeps running the cloud sync process. Because of that kubernetes cant detect the container failure and can't restart the container.
As a temporary fix i added a liveness probe that checks if java is running inside the container.
A more permanent solution might be needed to ensure resiliency. One such possible solution is to move the sync logic into a sidecar container.

### PR DESCRIPTION
**Pull request by syncush**
added logs
and graceful exit
Closes #1 

related to #60

### CODE REVIEW

File: images/replication-job/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,12 @@
 FROM ubuntu:18.04
+ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && apt-get install -y \
-    python-pip \
     libz-dev zlib1g-dev \
     curl unzip \
```
Comment (CptSchnitz) [inline images/replication-job/Dockerfile]: is unzip needed?

File: images/replication-job/start.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    exit(exit_code)
+
+def handle_osmosis_success():
+    log.info('osmosis successfully created a replication')
+
+# osmosis tuning: https://wiki.openstreetmap.org/wiki/Osmosis/Tuning,https://lists.openstreetmap.org/pipermail/talk/2012-October/064771.html
+def build_osmosis_command(min_interval, iterations, validate_schema_version):
+    validate_schema_version = 'yes' if validate_schema_version else 'no'
+    return 'osmosis --replicate-apidb iterations={0} minInterval={1} host={2} database={3} user={4} password={5} validateSchemaVersion={6} --write-replication workingDirectory={7}'\
+        .format(iterations, min_interval, pg_host, pg_db, pg_user, pg_password, validate_schema_version, replication_directory)
```
Comment (CptSchnitz) [inline images/replication-job/start.py:31]: missing empty line between functions.

File: images/replication-job/start.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def run_minute_replication():
+    osmosis_command = build_osmosis_command(osmosis_min_interval, osmosis_iterations, osmosis_validate_schema_version)
+    run_command(osmosis_command, process_log.info, process_log.error, handle_osmosis_failure, handle_osmosis_success)
+
+def main():
+    # if not os.path.exists(replication_directory):
+    while True:
+        run_minute_replication()
+        sleep(60)
```
Comment (CptSchnitz) [inline images/replication-job/start.py]: should not be a loop as osmosis runs all the time.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

**Review by CptSchnitz — APPROVED** [review]

File: images/replication-job/start.py (hunk 2 above)
Comment (syncush) [reply images/replication-job/start.py:31]: done

File: images/replication-job/Dockerfile (hunk 1 above)
Comment (syncush) [reply images/replication-job/Dockerfile]: done

File: images/replication-job/start.py (hunk 3 above)
Comment (syncush) [reply images/replication-job/start.py]: done

File: images/replication-job/start.py [hunk 4]
```diff
@@ -35,15 +35,14 @@ def run_minute_replication():
...
 def main():
-
     log.info('{0} is up'.format(app_name))
     run_minute_replication()
 
 if __name__ == "__main__":
-    base_log_path = '/var/log'
+    base_log_path = os.path.join('/var/log', app_name)
     service_logs_path = '{0}/{1}.log'.format(base_log_path, app_name)
     osmosis_logs_path = '{0}/{1}.log'.format(base_log_path, 'osmosis')
```
Comment (CptSchnitz) [inline images/replication-job/start.py]: use `os.path.join`

Comment (CptSchnitz) [inline images/replication-job/start.py]: add a variable to osmosis

File: images/replication-job/start.py [hunk 5]
```diff
@@ -35,15 +35,14 @@ def run_minute_replication():
...
-    os.makedirs('{0}/{1}'.format(base_log_path, app_name), exist_ok=True)
-    log = JSONLogger('main-debug', config={'handlers': {'file': {'filename': service_logs_path}}}, additional_fields={'service': app_name, 'description': 'service logs'})
-    process_log = JSONLogger('main-debug', config={'handlers': {'file': {'filename': osmosis_logs_path}}}, additional_fields={'service': app_name, 'description': 'osmosis logs'})
+    os.makedirs(base_log_path, exist_ok=True)
+    log = generate_logger(app_name, log_level='INFO', handlers=[{'type': 'rotating_file', 'path': service_logs_path},{ 'type': 'stream', 'output': 'stderr' }])
+    process_log = generate_logger('osmosis', log_level='INFO', handlers=[{'type': 'rotating_file', 'path': osmosis_logs_path}, { 'type': 'stream', 'output': 'stderr' }])
```
Comment (CptSchnitz) [inline images/replication-job/start.py]: add a variable to osmosis

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: images/replication-job/start.py (hunk 5 above)
Comment (syncush) [reply images/replication-job/start.py]: done

File: images/replication-job/start.py (hunk 4 above)
Comment (syncush) [reply images/replication-job/start.py]: done

Comment (syncush) [reply images/replication-job/start.py]: done

File: images/replication-job/start.py [hunk 6]
```diff
@@ -39,10 +39,11 @@ def main():
     run_minute_replication()
 
 if __name__ == "__main__":
+    osmosis_name = 'osmosis'
     base_log_path = os.path.join('/var/log', app_name)
-    service_logs_path = '{0}/{1}.log'.format(base_log_path, app_name)
-    osmosis_logs_path = '{0}/{1}.log'.format(base_log_path, 'osmosis')
+    service_logs_path = os.path.join(base_log_path, app_name)
+    osmosis_logs_path = os.path.join(base_log_path, osmosis_name)
```
Comment (CptSchnitz) [inline images/replication-job/start.py]: `.log` is missing

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: images/replication-job/start.py (hunk 6 above)
Comment (syncush) [reply images/replication-job/start.py]: done

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**